### PR TITLE
Regression exp abbas embed

### PR DESCRIPTION
**Pull request by AbasKhan**
## PR Summary

🚨 Breaking Change: Retire the `annotator_training_pipeline` CLI and associated HF model wrappers in favour of the new `ml_filter train_with_embeddings` entry point that expects pre-computed embeddings.

💾 HDF5 Embedding Support: Introduces `EmbeddingDataset`, automated shard discovery, and config validation so the trainer can consume Datatrove-style embedding exports without custom glue code.

🧠 Regression Head Training: Adds an `EmbeddingRegressionModel` with an MLP head, optional warm-start initialisation, and a purpose-built regression config template to train downstream heads quickly.

📈 Training UX Improvements: Spearman-first metrics, early stopping, and shared logging setup improve feedback loops while keeping HF logging noise under control.

📘 Documentation & Config Cleanup: Removes the deprecated `is_regression` flag, documents the new metric naming expectations, and refreshes sample configs to match the embedding workflow.

Downstream users need to switch to the new CLI and supply HDF5 embeddings plus split-aware metric names when fine-tuning regression heads.

### CODE REVIEW

File: examples/inference_pipeline/scripts/run_inference_pipeline_example.py [hunk 1]
```diff
@@ -1,11 +0,0 @@
-import os
```
Comment (ajude2s) [inline examples/inference_pipeline/scripts/run_inference_pipeline_example.py]: We definitely need code and tests for loading the trained checkpoints and inference part.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ repos:
   rev: 23.9.1
   hooks:
   - id: black
-    language_version: python3.11
+    language_version: python3.10
```
[the remark this answers is not in the document]
Comment (AbasKhan) [reply .pre-commit-config.yaml]: This version here is just for the formatting with black, until April it was actually 3.10 in the main branch aswell . I had reverted back to be consistent with the main. But now since main also has 3.11, i will update it here aswell.

File: src/ml_filter/training/embedding_training_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+        run_name=cfg.training.wandb_run_name,
+    )
+
+
+def _init_regression_weights(model: EmbeddingRegressionModel):
+    """Same initialization logic as your existing code."""
+    with torch.no_grad():
+        if hasattr(model.head, "mlp") and len(model.head.mlp) >= 3:
+            final_layer = model.head.mlp[2]  # The Linear(1000 -> 1) layer
+            final_layer.weight.data *= 10.0  # Scale up weights
```
[the remark this answers is not in the document]
Comment (AbasKhan) [reply src/ml_filter/training/embedding_training_pipeline.py]: Hahahaha lolz, 😅
